src/libutils/oglapphelpers/oglapp.h

```diff
@@ -141,11 +152,59 @@ class OglApp
     unsigned int m_imageTexID;
 
     OpenGLBuilderRcPtr m_oglBuilder;
+};
+
+class ScreenApp: public OglApp
+{
+public:
+    ScreenApp() = delete;
+    ScreenApp(const char * winTitle, int winWidth, int winHeight);
 
+    ~ScreenApp();
+    void redisplay() override;
+    void printGLInfo() const noexcept override;
+private:
+    // Window identifier returned by glutCreateWindow.
+    int m_mainWin{ 0 };
 };
 
-typedef OCIO_SHARED_PTR<OglApp> OglAppRcPtr;
+#ifdef OCIO_HEADLESS_ENABLED
+
+#include <EGL/egl.h>
+
+class HeadlessApp: public OglApp
+{
+public:
+    HeadlessApp() = delete;
+    HeadlessApp(const char * winTitle, int winWidth, int winHeight);
+
+    ~HeadlessApp();
+
+    void printGLInfo() const noexcept override;
+    // Helper to print EGL info.
```

Comment:
The method `printEGLInfo()` could be protected as only `printGLInfo()` must be used.